packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -1216,7 +1271,16 @@ export class MergeTree {
         if (pendingSegmentGroup !== undefined) {
             const deltaSegments: IMergeTreeSegmentDelta[] = [];
             pendingSegmentGroup.segments.map((pendingSegment) => {
-                const overlappingRemove = !pendingSegment.ack(pendingSegmentGroup, opArgs);
+                const attributionKey = (opArgs.op.type === MergeTreeDeltaType.INSERT && this.options?.attribution?.track) ?
```

Comment:
attribution isn't used in the ack call. personally it would be nice if the below code moved there

Reply:
I think it should live there long-term. But I don't want to move it there yet, because the source of the attribution information is eventually going to be more complex than just the op-stream: once we support attribution of things other than insertion, the type of the object passed into ack basically needs to be some description of all of the attribution information associated with that op. Some of that is going to be injected by the application, and there could be multiple types of attribution that get updated with a single op. For example a copy-paste might need to pass an object which indicates:

1. The op-based attribution for this ack is sequence# 25
2. This was a cut-and-paste of 3 other segments, with corresponding sequence numbers 15, 10, and 12
3. I'm an application that tracks bolding of text differently from other attribution in some other channel. A portion of the copy-pasted text was bolded at seq 8.

additionally, attribution of non-insert operations needs to be able to deal with existing attribution information on the segment.

When we eventually support all of this we'll clearly need a lot more information than just the attribution key being passed into the call to `ack`. But I don't want to settle any of those details right now, and `ack` is technically in the public merge-tree API. So I'd rather just hard-code the policy here for now.